### ISSUE TITLE
[Langfuse Issue] ChatLiteLLM: metadata are not logged into langfuse

### ISSUE THREAD

**Issue by databill86** [BerriAI/litellm#4059]
### What happened?

Hello, 
I've been working with `litellm` and `langfuse` for quite a while, traces are being logged correctly with `litellm` as an openai proxy. But, with `ChatLiteLLM`, while the traces are logged, there is no metadata. Everytime I pass metadata through `model_kwargs`, it's not taken into account. 

I'm using:

```
Docker image: ghcr.io/berriai/litellm:main-v1.40.4
Docker image: ghcr.io/langfuse/langfuse:2.49.0
langchain-core==0.2.3
langchain-openai==0.1.8
langchain-community==0.2.1
litellm==1.40.4
langfuse==2.35.0
openai==1.30.5
```

Here is the full code: 

```
import os
from langchain.chat_models import ChatLiteLLM
from langchain.schema import HumanMessage

chat = ChatLiteLLM(
    api_base="http://0.0.0.0:4004",
    model="openai/SOLAR-10.7B-Instruct-v1.0",
    custom_llm_provider="openai",
    model_kwargs={
        "metadata": {
            "trace_user_id": "user-id2",  # set langfuse Trace User ID
            "session_id": "session-1",  # set langfuse Session ID
            "tags": ["tag1", "tag2"]
        }
  )
messages = [
    HumanMessage(
        content="what model are you"
    )
]
chat(messages)

```

### Relevant log output

I see the trace in langfuse, but the metadata passed with arguments are not taken into account.

This is the request being sent to litellm: 

```
POST http://0.0.0.0:4004/chat/completions HTTP/1.1

{
    "max_tokens": 256,
    "messages": [
        {
            "content": "what model are you",
            "role": "user"
        }
    ],
    "model": "openai/SOLAR-10.7B-Instruct-v1.0",
    "n": 1,
    "stream": false,
    "temperature": 1
}
```

**Comment by ghost** [comment]
> I see the trace in langfuse, but the metadata passed with arguments are not taken into account.

hey @databill86 i suspect this means langfuse isn't sending the metadata to the proxy 

can you share your server logs?

**Comment by ghost** [comment]
oh i see you need to send extra metadata like this 

```
from langchain.chat_models import ChatOpenAI
from langchain.prompts.chat import (
    ChatPromptTemplate,
    HumanMessagePromptTemplate,
    SystemMessagePromptTemplate,
)
from langchain.schema import HumanMessage, SystemMessage
import os 

os.environ["OPENAI_API_KEY"] = "anything"

chat = ChatOpenAI(
    openai_api_base="http://0.0.0.0:4000",
    model = "gpt-3.5-turbo",
    temperature=0.1,
    extra_body={
        "metadata": {
            "generation_name": "ishaan-generation-langchain-client",
            "generation_id": "langchain-client-gen-id22",
            "trace_id": "langchain-client-trace-id22",
            "trace_user_id": "langchain-client-user-id2"
        }
    }
)

messages = [
    SystemMessage(
        content="You are a helpful assistant that im using to make a test request to."
    ),
    HumanMessage(
        content="test from litellm. tell me why it's amazing in 1 sentence"
    ),
]
response = chat(messages)

print(response)
```

**Comment by ghost** [comment]
use the `extra_body` param to pass the values

**Comment by databill86** [comment]
Thank you. That worked. 

However, I got this message: 
```
python3.11/site-packages/langchain_core/utils/utils.py:159: UserWarning: WARNING! extra_body is not default parameter.
                extra_body was transferred to model_kwargs.
                Please confirm that extra_body is what you intended.
  warnings.warn(
```

It's just a warning, but I prefer to use tracing it as intented. I don't want it to break the next release. 

I tried with `model_kwargs` instead of `extra_body` but I got this error: 

```
raise validation_error
pydantic.v1.error_wrappers.ValidationError: 1 validation error for ChatOpenAI
__root__
  Parameters {'metadata'} should be specified explicitly. Instead they were passed in as part of `model_kwargs` parameter. (type=value_error)
```

Using metadata as parameter didn't get the trace to show as intended in langfuse, the parameters passed are not taken into account.

**Comment by ghost** [comment]
hi @databill86 closing issue as it works now 

following up further though: 

how are you passing in the extra_body param in case 2?

the parameter in model_kwargs should be `extra_body` not metadata.

**Comment by databill86** [comment]
Thank you! that worked too, without any warnings. 
I'm now using `ChatOpenAI` instead of `ChatLiteLLM`, it's much simpler, there is no confusion in the model names, no need to pass custom_llm_provider.

**Comment by databill86** [comment]
By the way, thank you again for `litellm`! it's a great tool.

I use it with `langfuse`, so it would really be helpful to add some documentation about logging, and the parameters we can pass in `model_kwargs`. 
I tried to look in the source code by debugging and making breakpoints, but it's a bit hard to follow. 
So it would be great to list all the possible parameters we can pass in `model_kwargs`, `extra_body`, `metadata`...

**Comment by databill86** [comment]
@krrishdholakia , I have another problem, and promise, this is my last one. 
I think it's another bug, so I created a new issue here: https://github.com/BerriAI/litellm/issues/4093

**Comment by ghost** [comment]
> it would really be helpful to add some documentation about logging, and the parameters we can pass in model_kwargs

Hey @databill86 what docs do you want here? (sample code / examples from your current usage would be great!) 

Here's our langfuse doc today - https://docs.litellm.ai/docs/observability/langfuse_integration#trace--generation-parameters

**Comment by databill86** [comment]
That's perfect, I think I did search enough.
